src/Core.UnitTests/GitEventsMonitorTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace SonarLint.VisualStudio.Core.UnitTests
+{
+    [TestClass]
+    public class GitEventsMonitorTests
+    {
+        [TestMethod]
+        public void HeadChanged_EventInvoked()
+        {
+            int counter = 0;
+
+            var fileSystemWatcher = new Mock<IFileSystemWatcher>();
+
+            var fileSystemWatcherFactory = new Mock<IFileSystemWatcherFactory>();
+            fileSystemWatcherFactory.Setup(f => f.FromPath("C:\\Some Path\\.git")).Returns(fileSystemWatcher.Object);
+
+            var testSubject = CreateTestSubject("C:\\Some Path", fileSystemWatcherFactory);
+
+            fileSystemWatcher.VerifySet(w => w.Filter = "HEAD", Times.Once);
+            fileSystemWatcher.VerifySet(w => w.EnableRaisingEvents = true, Times.Once);
+
+            testSubject.HeadChanged += (o, e) => counter++;
+
+            fileSystemWatcher.Raise(w => w.Changed += null, null, null);
+
+            counter.Should().Be(1);
+        }
+
+        [TestMethod]
+        public void HeadChanged_MonitorDisposed_FileSystemWatcherDisposed()
+        {
+            var fileSystemWatcher = new Mock<IFileSystemWatcher>();
+
+            var fileSystemWatcherFactory = new Mock<IFileSystemWatcherFactory>();
+            fileSystemWatcherFactory.Setup(f => f.FromPath("C:\\Some Path\\.git")).Returns(fileSystemWatcher.Object);
+
+            var testSubject = CreateTestSubject("C:\\Some Path", fileSystemWatcherFactory);
+
+            fileSystemWatcher.VerifySet(w => w.Filter = "HEAD", Times.Once);
+            fileSystemWatcher.VerifySet(w => w.EnableRaisingEvents = true, Times.Once);
```

Comment:
Copy-and-paste-itis: we don't need to check the file watcher setup in every single test.
It's just clutter and detracts from the real purpose of the test.